### ISSUE TITLE
[API] - Name is required for user update

### ISSUE THREAD

**Issue by jdelucaa** [Unleash/unleash#1717]
### Describe the bug

Hello 👋 
According to the API [docs](https://docs.getunleash.io/api/admin/user-admin#add-a-new-user), the only properties that are required when creating a User (POST) are the `rootRole` and the `email` or `name` (by the way, the docs are wrong the `email` or `username` is required, the `username` is missing in the docs 🙈 and the property `name` is not really required). 

So I can create a User with the following payload:

POST admin/user-admin
```
{
    "username": "foo",
    "rootRole": "Admin"
}
```

But if I try to do a PUT (user update) and try to change the `rootRole` for example, with the following payload:

PUT admin/user-admin/1
```
{
    "username": "foo",
    "rootRole": "Editor"
}
```

I get the following error:
Status code: 400 Bad Request
```
[
    {
        "msg": "Empty .update() call detected! Update data does not contain any values to update. This will result in a faulty query. Table: users. Columns: name,username,email,image_url."
    }
]
```

The error is gone as soon as I provide the `name` property in the payload:

PUT admin/user-admin/1
```
{
   "name": "joana"
    "username": "foo",
    "rootRole": "Editor"
}
```
Response: 200

### Steps to reproduce the bug

See bug description.

### Expected behavior

I expect the required fields to be the same when creating and updating the users and that the documentation would reflect the actual behavior of the API.

### Logs, error output, etc.

```bash
[
    {
        "msg": "Empty .update() call detected! Update data does not contain any values to update. This will result in a faulty query. Table: users. Columns: name,username,email,image_url."
    }
]
```

### Unleash version

4.10.5

### Subscription type

Open source

### Hosting type

Self-hosted

**Comment by jdelucaa** [comment]
Another thing I noticed is that the `email` is not required for POST (if you provide the `username`), but it is required for PUT (even if you provide the username), better saying: you cannot remove the email after you add it, but you can create a user without email.

**Comment by andreas-unleash** [comment]
Hi, thank you for bringing this to our attention.  I will raise it internally with the team.

**Comment by thomasheartman** [comment]
Hey, @jdelucaa! 👋🏼  Thanks again for raising this. You are perfectly right in what you're saying. I'll get on it and update the documentation right away. at the same time, the API behavior in this case seems kind of strange (that you need to provide a `name` even if there isn't one there already). I suspect we'll want to change that somehow.

I'll get back to you with more news later. In the meantime: we _are_ working on integrating OpenAPI into all the endpoints, so that should prevent errors like this from popping up again 🤞🏼

**Comment by thomasheartman** [comment]
@jdelucaa I've added a PR that fixes the docs now (feel free to have a look and see if you agree: #1727). However, I think I'd like to keep this issue open until we figure out whether we want to do something about this endpoint. You're quite right that the way it works at the moment is very confusing and doesn't seem to make a lot of sense to me.

**Comment by jdelucaa** [comment]
> @jdelucaa I've added a PR that fixes the docs now (feel free to have a look and see if you agree: #1727). However, I think I'd like to keep this issue open until we figure out whether we want to do something about this endpoint. You're quite right that the way it works at the moment is very confusing and doesn't seem to make a lot of sense to me.

Hi @thomasheartman , thanks for the quick reply! The changes you made to the docs look good to me! Looking forward for the improvements in the API! Thanks a lot! 🚀

**Comment by thomasheartman** [comment]
We've fixed this in #1730 now, so I'll go ahead and close this now 🙋 The changes should be released in one of the very next versions of Unleash. Feel free to reopen this issue if you feel that there's more work to be done.

Thanks again for your report and your input: it's very appreciated! 🙏
